### PR TITLE
Localize registries.

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: src/vcpkg/registries.cpp [hunk 1]
```diff
@@ -723,29 +720,36 @@ namespace
...
-                    "pull`.\nAvailable versions:\n",
-                    Strings::join("", port_versions, [](const Version& v) { return Strings::concat("    ", v, "\n"); }),
-                    "\nSee `vcpkg help versioning` for more information."),
-                expected_right_tag};
+            return format_version_git_entry_missing(port_name, version, port_versions);
         }
 
         const auto& git_tree = git_trees[it - port_versions.begin()];
         return m_paths.git_checkout_port(port_name, git_tree, m_paths.root / ".git")
+            .map_error([](std::string&& err) { return LocalizedString::from_raw(err); })
```
Comment (BillyONeal) [inline src/vcpkg/registries.cpp:752]: Gone after https://github.com/microsoft/vcpkg-tool/pull/914

File: src/vcpkg/registries.cpp [hunk 2]
```diff
@@ -792,20 +797,13 @@ namespace
...
-                    Strings::join("", port_versions, [](const Version& v) { return Strings::concat("    ", v, "\n"); }),
-                    "\nSee `vcpkg help versioning` for more information."),
-                expected_right_tag};
+            return format_version_git_entry_missing(port_name, version, port_versions);
         }
 
         const auto& git_tree = git_trees[it - port_versions.begin()];
-        return parent.m_paths.git_checkout_object_from_remote_registry(git_tree).map(
+        return parent.m_paths.git_checkout_object_from_remote_registry(git_tree)
+            .map_error([](std::string&& err) { return LocalizedString::from_raw(err); })
```
Comment (BillyONeal) [inline src/vcpkg/registries.cpp:805]: Gone after https://github.com/microsoft/vcpkg-tool/pull/914

File: src/vcpkg/registries.cpp [hunk 3]
```diff
@@ -722,30 +719,35 @@ namespace
...
-                    "Error: No version entry for ",
-                    port_name,
-                    " at version ",
-                    version,
-                    ". This may be fixed by updating vcpkg to the latest master via `git "
-                    "pull`.\nAvailable versions:\n",
-                    Strings::join("", port_versions, [](const Version& v) { return Strings::concat("    ", v, "\n"); }),
-                    "\nSee `vcpkg help versioning` for more information."),
-                expected_right_tag};
+            return format_version_git_entry_missing(port_name, version, port_versions);
```
Comment (ras0219-msft) [inline src/vcpkg/registries.cpp:746]: For the `BuiltinGitRegistry`, we should suggest updating the vcpkg instance. This registry type requires user action to make new versions available.

File: src/vcpkg/registries.cpp [hunk 4]
```diff
@@ -868,50 +861,47 @@ namespace
...
 
         auto maybe_versions_json = Json::parse(contents);
-        if (!maybe_versions_json)
+        auto versions_json = maybe_versions_json.get();
+        if (!versions_json)
         {
-            return Strings::format("Error: failed to parse versions file for `%s`: %s",
-                                   port_name,
-                                   maybe_versions_json.error()->to_string());
+            return msg::format_error(msgFailedToParseJson, msg::path = versions_file_path);
```
Comment (ras0219-msft) [inline src/vcpkg/registries.cpp:893]: This no longer emits the detailed parse errors. This is especially important for Filesystem registries where the user might be actively editing them.

**Review by ras0219-msft — APPROVED** [review]

File: src/vcpkg/registries.cpp (hunk 4 above)
Comment (BillyONeal) [reply src/vcpkg/registries.cpp:893]: I think basically all of the "msgFailedToParseJson" ones are errors. Stand by...

Comment (BillyONeal) [reply src/vcpkg/registries.cpp:893]: Example: 
image

Comment (BillyONeal) [reply src/vcpkg/registries.cpp:893]: image

File: src/vcpkg/registries.cpp (hunk 3 above)
Comment (BillyONeal) [reply src/vcpkg/registries.cpp:746]: For once I like the upgrade message! Agreed!

File: src/vcpkg/registries.cpp (hunk 4 above)
Comment (BillyONeal) [reply src/vcpkg/registries.cpp]: Example:

image

**Pull request merged**